src/main/kotlin/io/snyk/plugin/SnykBulkFileListener.kt

```diff
@@ -79,9 +86,20 @@
                     .filter { iacFileExtensions.contains(it.extension) }
                     .find { ProjectRootManager.getInstance(project).fileIndex.isInContent(it) }
                 if (iacRelatedFileChanged != null) {
-                    toolWindowPanel.currentIacResult = null
-                    LOG.debug("IaC cached results dropped due to changes in: $iacRelatedFileChanged")
-                    DaemonCodeAnalyzer.getInstance(project).restart()
+                    val iacTreeRoot = toolWindowPanel.getRootIacIssuesTreeNode()
+                    val nodes = iacTreeRoot.depthFirstEnumeration()
+                    while (nodes.hasMoreElements()) {
+                        val node = nodes.nextElement() as DefaultMutableTreeNode
+                        if (node is IacFileTreeNode) {
+                            makeIacTreeNodeObsolete(
```

Comment:
code above will make obsolete only first IaC related file changed. It could be few of them here due to batch change operation, like `git checkout`

Reply:
Oh wow, yeah, that's an unintended bug. Good catch.